src/components/UserInputBlock.tsx

```diff
@@ -51,12 +66,66 @@ export const UserInputBlock = ({ content, prompt, ack, waitForEvent, fireEvent,
       .max(MAX_CHARACTER_INPUT_BLOCK_LENGTH, EXCEEDED_MAX_INPUT_ERROR)
   })
 
+  const updateInteractiveState = async (): Promise<void> => {
+    try {
+      if (contentId === undefined || persistor === undefined) {
+        setInteractiveState({
+          userResponse: '',
+          responseSubmitted: false
+        }
+        )
+        setPersistorStatus(PersistorStatus.Success)
+        return
+      }
+
+      const currentLocalStorageState = await persistor.get(contentId)
+      if (currentLocalStorageState !== null) {
+        const parsedLocalStorage = JSON.parse(currentLocalStorageState)
+        setInteractiveState({
+          userResponse: parsedLocalStorage.userResponse,
+          responseSubmitted: parsedLocalStorage.responseSubmitted
+        })
+        setResponseSubmitted(parsedLocalStorage.responseSubmitted)
+      } else {
+        setInteractiveState({
+          userResponse: '',
+          responseSubmitted: false
+        })
+      }
+      setPersistorStatus(PersistorStatus.Success)
+    } catch (err) {
+      setPersistorStatus(PersistorStatus.Failure)
+      console.error(err)
+    }
+  }
+
+  useEffect(() => {
+    updateInteractiveState().catch(() => { })
+  }, [])
+
   const handleSubmit = async (values: InputFormValues, { setFieldError }: FormikHelpers<InputFormValues>): Promise<void> => {
     if (values.response.trim() === '') {
       setFieldError('response', NON_EMPTY_VALUE_ERROR)
       return
     }
 
+    try {
+      if (contentId !== undefined && persistor !== undefined) {
+        await persistor.put(contentId, JSON.stringify({ userResponse: values.response, responseSubmitted: true }))
```

Comment:
If we want to use a type definition to enforce our desired schema for persisted data, we may want to do something like:

```javascript
        const newPersistedData: PersistedData = { userResponse: values.response, responseSubmitted: true }
        await persistor.put(contentId, JSON.stringify(newPersistedData))
```